src/init.cpp

```diff
@@ -1276,6 +1276,30 @@ bool AppInitMain(NodeContext& node, interfaces::BlockAndHeaderTipInfo* tip_info)
                                      chainman, *node.mempool, ignores_incoming_txs);
     RegisterValidationInterface(node.peerman.get());
 
+    // Check port numbers
+    const bilingual_str port_error = _("Invalid port specified in %s: %s");
+    const std::string port_options_int[] = {"-port", "-rpcport"};
+    for (const std::string& port_option : port_options_int) {
+        if (args.IsArgSet(port_option)) {
+            int64_t portIn = args.GetIntArg(port_option, 0);
+            if (portIn <= 0 || portIn > std::numeric_limits<uint16_t>::max())
+                return InitError(strprintf(port_error, port_option, ToString(portIn)));
```

Comment:
nit: I don't think this needs ToString (strprintf should handle int64_t as `%s` just fine)

```suggestion
                return InitError(strprintf(port_error, port_option, portIn));
```